### PR TITLE
PopoverWithRef, plus some other stuff

### PR DESCRIPTION
**Pull request by npfitz**
### Description
Migrates `<ClickActionsPopover >` to use Mantine popovers, as well as adds some building blocks for future migrations away from Tippy and Legacy popover components

### How to verify

1. Go to any table, and click the header. It should work as it always does

Current sadness - Clicking the header again moves the popover for a couple cycles before computing the correct spot again.

### Demo
image

### Checklist

- [ ] ~Tests have been added/updated to cover changes in this PR~

### CODE REVIEW

**Comment by rafpaf** [comment]
@npfitz I unpublished this, hope that was OK

File: frontend/src/metabase/visualizations/components/ClickActions/ClickActionsPopover.tsx [hunk 1]
```diff
@@ -136,7 +146,9 @@ export class ClickActionsPopover extends Component<
         ) : (
           <ClickActionsView
             clickActions={clickActions}
-            close={this.close}
+            close={() => {
+              this.close();
+            }}
```
Comment (iethree) [inline frontend/src/metabase/visualizations/components/ClickActions/ClickActionsPopover.tsx:151]: is this necessary?

**Comment by iethree** [comment]
> @npfitz I unpublished this, hope that was OK

Why? do you have specific misgivings?

**Review by iethree — APPROVED** [review]
Very nice! the only thing I might add is a comment explaining that this component isn't something we should reach for except to make it easier to factor out legacy popovers.

**Comment by npfitz** [comment]
> Why? do you have specific misgivings?
I wasn't sure how we wanted to proceed with this, so I kinda implied to @rafpaf that this may not be ready for prime time haha. Happy to move forward with it though now that appears to be working okay.

**Review by iethree — APPROVED** [review]

**Pull request merged**